src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -161,7 +161,16 @@ private static void ValidateTimestampCms(SigningSpecifications spec, SignedCms t
 
             if (!CertificateUtility.IsSignatureAlgorithmSupported(signerInfo.Certificate))
             {
-                throw new TimestampException(NuGetLogCode.NU3022, Strings.SignError_TimestampUnsupportedSignatureAlgorithm);
+                var certificateSignatureAlgorithm = GetNameOrOidString(signerInfo.Certificate.SignatureAlgorithm);
+
+                var supportedSignatureAlgorithms = string.Join(", ", spec.AllowedCertificateSignatureAlgorithms);
+
+                var errorMessage = string.Format(CultureInfo.CurrentCulture,
+                    Strings.SignError_TimestampCertificateUnsupportedSignatureAlgorithm,
```

Comment:
I'd remove "SignError_" here and below.  This error should apply at signing time as well as verification time.  "SignError_" doesn't add much information.